### ISSUE TITLE
Onboarding: Onboarding settings for issuers

### ISSUE THREAD

**Issue by sophialittlejohn** [centrifuge/apps#1311]
- [x] Enable/disable POD read option - access to private asset-level data: [info from slack](https://kflabs.slack.com/archives/C04GARZ532T/p1685009054706059?thread_ts=1685008799.794619&cid=C04GARZ532T)
- [x] Enable/disable investor accreditation for US investors
- [x] Change design to upload subscription agreements 
- [x] Restrict onboarding from different countries
- [x] Allow external onboarding providers (less important)

The goal is to set up the frontend and pool config in a way that would allow issuers to choose their own onboarding provider. 

The pool config could look something like this:

```
{
    ...,
    "onboarding": {
        "type": "centrifuge" | "external" | "none",
        "external": { // only relevant if type == external
            "link": "https:/..."
        },
        "centrifuge": { // only relevant if type == centrifuge
            "agreements: {
                [TRANCHE_ID]: {
                    "template": "text", // ipfs hash,
                    "country": "us | non-us",
                    "target": "individual | entity"
                }
            },
            "restrictedCountryCodes": [], // list of 2 letter country codes which are not allowed to onboard to this pool
            "accreditationVerificationMethod": "self-attestation" | "document-verification" | "none",
        }
    }
}
```

See conversation here: https://centrifugedao.slack.com/archives/C04HFEXBX8B/p1676996182456149

## Design
[Figma file](https://www.figma.com/file/9iwvzHuvbjlHXRWlPWRxxc/Centrifuge-app---Version-2-(Branch)?type=design&node-id=9464-61672&t=aTSjmSuuhiGOeTWE-0)

<img width="529" alt="Screenshot 2023-05-24 at 16 09 16" src="https://github.com/centrifuge/apps/assets/31280003/157a7256-01d8-4ce7-a7c1-2afb245930cf">

**Comment by sophialittlejohn** [comment]
> Enable/disable investor accreditation for US investors
Before I build this I want to make sure I understood it properly. From my understanding if you are a US investor you legally MUST be accredited. Is the idea here that issuers can choose if they want to allow US based entities/individuals to onboard at all? And if they choose to allow US investors, issuers can choose if they investors should do a self-attestation or provide documents?

>  Change design to upload subscription agreements
Done! However, I built it using our existing file upload component from fabric so that it matches the rest of the file uploads.

> Restrict onboarding from different countries
I've built a version of this but had to modify it a bit. Natively the select component does not support search functionality. Instead I used the browser native element `datalist`. Also, shuftipro supports different countries for KYC and KYB, so I split them into two inputs and two lists that can be managed individually. One to choose restricted countries for KYB and one to choose restricted countries for KYC.

> Allow external onboarding providers (less important)
Since there is only one option for now, I don't think it makes sense to build it now. I would suggest revisiting when it's becomes relevant.

Hope the decisions made are ok 😄

**Comment by offerijns** [comment]
Since Anna is out I will jump in here :)

> Is the idea here that issuers can choose if they want to allow US based entities/individuals to onboard at all? And if they choose to allow US investors, issuers can choose if they investors should do a self-attestation or provide documents?

This is both correct, yeah.

> Since there is only one option for now, I don't think it makes sense to build it now. I would suggest revisiting when it's becomes relevant.

I think we should already offer the three options, 1) no automated onboarding, 2) Centrifuge onboarding, 3) external. If they choose external, the "Onboard to X" button will just redirect the user to that external link. This way an issuer could technically already build their own onboarding flow.

I also think this is important to build for another reason: the default option should be 1. If an issuer toggles 2 here, the transaction to update the pool config can then be batched with the tx to add the pure proxy as a `MemberlistAdmin` to the pool.

**Comment by sophialittlejohn** [comment]
Thanks @offerijns 😄  can defs work on the accredited investor section.

> I also think this is important to build for another reason: the default option should be 1. If an issuer toggles 2 here, the transaction to update the pool config can then be batched with the tx to add the pure proxy as a MemberlistAdmin to the pool.

Good point why we should build this now. 

Say an issuer wants to provide their own onboarding link. I think we need a place or them to specify it. For example when they select `external` from the dropdown we provide a field for the link they'd like to use.

When an issuer selects none I will hide the onboarding button for investors. 

If the default is 1 (centrifuge onboarding), would it not make more sense to include the pure proxy whitelisting in pool creation and remove it if the issuer chooses a different onboarding method?

**Comment by offerijns** [comment]
> But I will not build a select yet to choose what kind of attestation unless you see a reason why we should build it now already.

Definitely that makes sense

> Say an issuer wants to provide their own onboarding link. I think we need a place or them to specify it. For example when they select external from the dropdown we provide a field for the link they'd like to use.

Also a good point!

> If the default is 1 (centrifuge onboarding), would it not make more sense to include the pure proxy whitelisting in pool creation and remove it if the issuer chooses a different onboarding method?

I would make the default "none". Because the Centrifuge onboarding anyways won't work until also the other parts of the onboarding settings are also done (uploaded the agreements). So otherwise there is authorization to the onboarding API without it actually being functional.

**Comment by sophialittlejohn** [comment]
> Is it possible to show unsupported countries in the list below and disable the option to remove them instead of showing the dropdown and to remove them from the dropdown list?

I can do that, just note that will be a long list by default then for both KYC and KYB!

Regarding the POD option, the text implies that we can automatically give investors POD read access (when whitelisting), but actually we can't. To give pool specific permissions (POD read access) requires the pool admin to sign the transaction... So if this is important we need some kind of UX for issuers to click and sign the tx 😄

**Comment by offerijns** [comment]
> Regarding the POD option, the text implies that we can automatically give investors POD read access (when whitelisting), but actually we can't.

The pure proxy is a `MemberListAdmin` and is able to both add `TrancheInvestor` as well as `PODReadAccess` permissions to members.

**Comment by sophialittlejohn** [comment]
thanks! it appears we will need to wait on this ticket to be tackled by protocol before

**Comment by sophialittlejohn** [comment]
@annamehr are you sure we need to upload exec summary and agreements together? Currently exec summaries are part of the pool configuration. They can be added when creating a pool or after in `issuer/<pool-id>/configuration`.  AFAIK combing exec summaries and agreements was only a requirement for tinlake pools - this settings page is only available for centrifuge pools.

**Comment by annamehr** [comment]
@sophialittlejohn you're correct exec summaries are uploaded on pool creation. I created this new ticket for now and asked Tim to work on an enhanced design then we can close this ticket for now. #1449
